cmd/access/unstaked_access_node_builder.go

```diff
@@ -93,22 +100,35 @@ func (builder *UnstakedAccessNodeBuilder) enqueueUnstakedNetworkInit() {
 		libP2PFactory, err := builder.FlowAccessNodeBuilder.initLibP2PFactory(unstakedNodeID, unstakedNetworkMetrics, unstakedNetworkKey)
 		builder.MustNot(err)
 
-		// use the default validators for the staked access node unstaked networks
-		msgValidators := p2p.DefaultValidators(builder.Logger, unstakedNodeID)
+		msgValidators := unstakedNetworkValidators(unstakedNodeID)
 
 		// don't need any peer updates since this will be taken care by the DHT discovery mechanism
 		peerUpdateInterval := time.Hour
 
-		middleware := builder.initMiddleware(unstakedNodeID, unstakedNetworkMetrics, libP2PFactory, peerUpdateInterval, msgValidators...)
-
-		// empty list of unstaked network participants since they will be discovered dynamically and are not known upfront
-		participants := flow.IdentityList{}
+		middleware := builder.initMiddleware(unstakedNodeID, unstakedNetworkMetrics, libP2PFactory, peerUpdateInterval,
+			false, // no connection gating for the unstaked network
+			false, // no peer management for the unstaked network (peer discovery will be done via LibP2P discovery mechanism)
+			msgValidators...)
 
 		upstreamANIdentifier, err := flow.HexStringToIdentifier(builder.stakedAccessNodeIDHex)
 		builder.MustNot(err)
 
+		nodes, err := builder.State.Sealed().Identities(filter.HasNodeID(upstreamANIdentifier))
+		builder.MustNot(err)
+
+		stakedAN := nodes[0]
+
+		modifiedStakedAN := &flow.Identity{
+			NodeID:        stakedAN.NodeID,
+			Address:       builder.stakedAccessNodeAddress,
+			Role:          stakedAN.Role,
+			NetworkPubKey: stakedAN.NetworkPubKey,
+		}
```

Comment:
Question: why would the `stakedAccessNodeAddress` be different from `stakedAN.Address`?